robot/robot.py

```diff
@@ -167,7 +174,14 @@ def mode(self):
         """
         return self._game.mode
 
-    def __del__(self):
+    def close(self):
         # stop the polling threads
         for camera in self.known_cameras:
             camera._stop_poll()
+        for board_type in self.all_known_boards:
+            for board in board_type:
+                board._clean_up()
```

Comment:
I'd suggest renaming `_clean_up` to use the `__del__` method, so it's automatically called when a board is destroyed, and means we're not adding additional custom functionality

Reply:
I agree, although probably worth also doing an explicit `del board` too.